### PR TITLE
samples: net: socket: packet: Refactor to allow flooding

### PR DESCRIPTION
**Pull request by jukkar**
Create two threads, one for receiving packet socket data and
the other for sending raw Ethernet frames.
Add flood option where it is possible to stress test the IP
stack. Flooding is disabled by default.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

### CODE REVIEW

File: samples/net/sockets/packet/Kconfig [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+# Copyright (c) 2020 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+mainmenu "Networking packet socket sample application"
+
+config NET_SAMPLE_ENABLE_FLOODING
+	bool "Send packets as fast as possible"
+	help
+	  This will stress test the network stack. By default
```
Comment (pfalcon) [inline samples/net/sockets/packet/Kconfig:11]: The description here is confusing. Does enabling this option causes packets to be sent every 5s? That contradicts the title of the option "Send packets as fast as possible". And if it's not the case, why it's written in that (confusing) way in the description of this option.

Suggested: "If this option not enabled, a packet is sent every 5 seconds". But then, why not add a numeric option, delay in ms, where 0 would mean "flood" ?

Comment (jukkar) [reply samples/net/sockets/packet/Kconfig:11]: > The description here is confusing.

What is confusing there. The default value is 'n' which means that a packet is sent every 5 seconds.

Comment (jukkar) [reply samples/net/sockets/packet/Kconfig:11]: Allowing user to supply the timeout makes sense, I will implement that.

**Review by pfalcon — APPROVED** [review]

File: samples/net/sockets/packet/src/packet.c [hunk 2]
```diff
@@ -90,71 +95,91 @@ static int start_packet_socket(void)
...
+				sizeof(packet->recv_buffer), 0);
+
+		if (received < 0) {
+			LOG_ERR("RAW : recv error %d", errno);
+			ret = -errno;
+			break;
+		}
+
+		LOG_DBG("Received %d bytes", received);
+
```
Comment (rlubos) [inline samples/net/sockets/packet/src/packet.c:116]: nit: Unneeded empty line.

**Review by rlubos — APPROVED** [review]

**Pull request merged**